core/invoker/src/main/scala/whisk/core/container/ContainerPool.scala

```diff
@@ -422,25 +430,31 @@ class ContainerPool(
 
     /* A background thread that
      *   1. Kills leftover action containers on startup
-     *   2. Periodically re-populates the container pool with fresh (un-instantiated) nodejs containers.
+     *   2. (actually a Future) Periodically re-populates the container pool with fresh (un-instantiated) nodejs containers.
      *   3. Periodically tears down containers that have logically been removed from the system
-     *   4. Each of the significant method subcalls are gaurded to not throw an exception.
+     *   4. Each of the significant method subcalls are guarded to not throw an exception.
      */
     private def nannyThread(allContainers: Seq[ContainerState]) = new Thread {
         override def run {
-            implicit val tid = TransactionId.invokerWarmup
+            implicit val tid = TransactionId.invokerNanny
             if (!standalone) {
                 killStragglers(allContainers)
-            }
-            while (true) {
-                Thread.sleep(100) // serves to prevent busy looping
-                // create a new stem cell if the number of warm containers is less than the count allowed
+                // Create a new stem cell if the number of warm containers is less than the count allowed
                 // as long as there is slack so that any actions that may be waiting to create a container
                 // are not held back; Note since this method is not fully synchronized, it is possible to
                 // start this operation while there is slack and end up waiting on the docker lock later
-                if (!standalone && getNumberOfIdleContainers(warmNodejsKey) < WARM_NODEJS_CONTAINERS && slack() > 0) {
-                    addWarmNodejsContainer()(tid)
+                val warmupInterval = 100.milliseconds
+                Scheduler.scheduleWaitAtLeast(warmupInterval) { () =>
+                    implicit val tid = TransactionId.invokerWarmup
+                    if (getNumberOfIdleContainers(warmNodejsKey) < WARM_NODEJS_CONTAINERS && slack() > 0 && toBeRemoved.size < RM_THRESHOLD) {
+                        addWarmNodejsContainer()(tid)
+                    } else {
+                        Future.successful(())
+                    }
                 }
+            }
+            while (true) {
+                Thread.sleep(100) // serves to prevent busy looping
```

Comment:
1.  Possibly but there would only be one message so it seems more like a plain periodic task to me.
2.  For now, I want the removal operation to be prioritized as high as possible and not be part of the Future scheduling so it is guaranteed dedicated access.   We can revisit this once the current issue has been handled.
3. It cannot be off the main thread (during construction) because killStragglers can be slow and we need the service to start up.  However, it could be moved to a future of its own though this would still have to sequentially precede the others as killStragglers must run to completion first.